nixos/modules/services/system/cachix-watch-store.nix

```
@@ -0,0 +1,77 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cachix-watch-store;
+in
+{
+  meta.maintainers = [ lib.maintainers.jfroche ];
+
+  options.services.cachix-watch-store = {
+    enable = mkEnableOption (lib.mdDoc "Cachix Watch Store: https://docs.cachix.org");
+
+    cacheName = mkOption {
+      type = types.str;
+      description = lib.mdDoc "Cachix binary cache name";
+    };
+
+    compressionLevel = mkOption {
+      type = types.nullOr types.int;
+      description = lib.mdDoc "The compression level for XZ compression (between 0 and 9)";
+      default = null;
+    };
+
+    jobs = mkOption {
+      type = types.nullOr types.int;
+      description = lib.mdDoc "Number of threads used for pushing store paths";
+      default = null;
+    };
+
+    host = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = lib.mdDoc "Cachix host to connect to";
+    };
+
+    verbose = mkOption {
+      type = types.bool;
+      description = lib.mdDoc "Enable verbose output";
+      default = false;
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.cachix;
+      defaultText = literalExpression "pkgs.cachix";
+      description = lib.mdDoc "Cachix Client package to use.";
+    };
+
+    configFile = mkOption {
+      type = types.path;
+      description = lib.mdDoc ''
+        Required file that needs to contain the cachix configuration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.cachix-watch-store-agent = {
+      description = "Cachix watch store Agent";
+      after = [ "network-online.target" ];
+      path = [ config.nix.package ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        # we don't want to kill children processes as those are pushing packages
+        KillMode = "process";
+        Restart = "on-failure";
+        DynamicUser = true;
+        LoadCredential = [
+          "config:${cfg.configFile}"
+        ];
+        ExecStart = ''
+          ${cfg.package}/bin/cachix --config ''${CREDENTIALS_DIRECTORY}/config${lib.optionalString cfg.verbose " --verbose"}${lib.optionalString (cfg.host != null) " --host ${cfg.host}"} watch-store${lib.optionalString (cfg.compressionLevel != null) " --compression-level ${cfg.compressionLevel}"}${lib.optionalString (cfg.jobs != null) " --jobs ${toString cfg.jobs}"} ${cfg.cacheName}'';
```

```suggestion
        ExecStart =
          let command = [''${cfg.package}/bin/cachix" "--config" "'${CREDENTIALS_DIRECTORY}/config"] ++ (lib.optional cfg.verbose "--verbose") ++ (lib.optionals (cfg.host != null) ["--host" cfg.host]) ++ ["watch-store"] ++ (lib.optionals cfg.compressionLevel != null) ["--compression-level" (toString cfg.compressionLevel)]) ++ (lib.optionals (cfg.jobs != null) ["--jobs" (toString cfg.jobs)]) ++ [cfg.cacheName];
          in
            lib.escapeShellArgs command;
```
Something like this could be more readable?